### DISCUSSION TITLE
Send email to reset password on first attempt to access Keycloak

### DISCUSSION THREAD

**Discussion by SilvandroOliveira** [keycloak/keycloak#21471] [Q&A]
We are migrating our native authentication functionality to Keycloak and our user base has a password encrypted in SHA256, I will use the Keycloak API to load users, but without the password, I would like to know if I can somehow do it on the first login the user to be automatically sent an email with a link to reset the password.

**Comment by sohaibaumer** [comment]
You can send an email with a password reset link to users on their first login to Keycloak. 

- Create a custom authenticator in Keycloak.   
- Integrate with an email service to send the email.
- Modify the Keycloak authentication flow to include the custom authenticator.
- Trigger the email sending process in the custom authenticator.
- Set up a password reset page or endpoint in your application.
